### DISCUSSION TITLE
Help adapting SQLAlchemy engine for Linux

### DISCUSSION THREAD

**Discussion by andagi10** [sqlalchemy/sqlalchemy#12039] [Usage Questions]
Hello everyone,

I’ve written a Python program that creates a table using SQLAlchemy, and it’s been working well on a Windows computer. Now, I’d like to run the same code on my Linux machine, but I’m not sure how to adjust my SQLAlchemy engine for compatibility.

Could you provide some advice on what changes might be necessary? For instance, should I look into specific connection strings or drivers that work better on Linux?

Any tips or best practices for setting up SQLAlchemy on Linux would be greatly appreciated. Thanks in advance!

**Comment by CaselIT** [comment]
Hi,

No change should be required. Is something not working?

**Reply by andagi10** [reply]
Hi,

Thank you for your response! The program works fine on Windows, but on Linux, I’m getting an error message:

SSPI: Provider: No credentials were supplied

![image](https://github.com/user-attachments/assets/714312a4-836e-42b0-a59b-593dcb928a66)


From what I understand, SSPI is specific to Windows authentication. Do you know if there’s an equivalent setup or workaround for Linux? Any tips on configuring authentication on Linux with SQLAlchemy would be really helpful!

Thanks again!

**Reply by gordthompson** [reply]
Are you using `Trusted_Connection=yes` ?

**Reply by andagi10** [reply]
> Are you using `Trusted_Connection=yes` ?

I have tried that but now i am only using TrustServerCertificate=Yes&Authentication=SqlPassword

**Reply by gordthompson** [reply]
Verify that your target SQL Server instance supports SQL Server authentication (UID and PWD). The following link may prove helpful

https://www.mssqltips.com/sqlservertip/2191/how-to-check-sql-server-authentication-mode-using-t-sql-and-ssms/
